www/docs/server/non-json-content-types.md

```diff
@@ -0,0 +1,104 @@
+---
+id: non-json-content-types
+title: Non-JSON Content Types
+sidebar_label: Non-JSON Inputs (FormData, File, Blob)
+slug: /server/non-json-content-types
+---
+
+In addition to JSON-serializable data, tRPC can use FormData, File, and other Binary types as procedure inputs
+
+## Client Setup
+
+:::warning
+While tRPC natively supports several non-json serializable types, your client may need a little link configuration to support them depending on your setup.
+:::
+
+`httpLink` supports non-json content types out the box, if you're only using this then your existing setup should work immediately
+
+```ts
+import { httpLink } from '@trpc/client';
+
+trpc.createClient({
+  links: [
+    httpLink({
+      url: 'http://localhost:2022',
+    }),
+  ],
+});
+```
+
+However, not all links support these new content types, if you're using `httpBatchLink` or `httpBatchStreamLink` you will need to include a splitLink and check which link to use depending on the content
+
+```ts
+import {
+  httpBatchLink,
+  httpLink,
+  isNonJsonSerializable,
+  splitLink,
+} from '@trpc/client';
+
+trpc.createClient({
+  links: [
+    splitLink({
+      condition: (op) => isNonJsonSerializable(op.input),
+      true: httpLink({
+        url,
+      }),
+      false: httpBatchLink({
+        url,
+      }),
+    }),
+  ],
+});
+```
+
+## `FormData` Input
+
+FormData is natively supported, and for more advanced usage you could also combine this with a library like [zod-form-data](https://www.npmjs.com/package/zod-form-data) to validate inputs in a type-safe way.
```

Comment:
could be good to show that you don't need an external lib to do this?

https://github.com/juliusmarminge/trpc-interop/blob/main/src/server.ts#L28-L43

Reply:
I've added a permalink to that, it's definitely useful